exercises/forth/example_gen.go

```diff
@@ -0,0 +1,77 @@
+// +build ignore
+
+package main
+
+import (
+	"log"
+	"text/template"
+
+	"../../gen"
+)
+
+func main() {
+	t, err := template.New("").Parse(tmpl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var j js
+	if err := gen.Gen("forth", &j, t); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// The JSON structure we expect to be able to unmarshal into
+type js struct {
+	ParsingAndNumbers  section `json:"parsing and numbers"`
+	Addition           section
+	Subtraction        section
+	Multiplication     section
+	Division           section
+	CombinedArithmetic section `json:"combined arithmetic"`
+	Dup                section
+	Drop               section
+	Swap               section
+	Over               section
+	UserDefinedWords   section `json:"user-defined words"`
+}
+
+// A grouped section of test cases.
+type section struct {
+	Cases []struct {
+		Description string
+		Input       []string
+		Expected    []int
+	}
+}
+
+// template applied to above data structure generates the Go test cases
+var tmpl = `package forth
+
+// Source: {{.Ori}}
+{{if .Commit}}// Commit: {{.Commit}}
+{{end}}
+
+{{/* template for repeated caseSection */}}
+
+{{define "caseSection"}}
+{{range $y, $c := .}}{
+	{{printf "%q" $c.Description}},
+	{{printf "%#v" $c.Input}},
+	{{printf "%#v" $c.Expected}},
+},
+{{end}}{{end}}
+
+var testCases = []testCase{
+// parsing and numbers{{template "caseSection" .J.ParsingAndNumbers.Cases}}
+// addition{{template "caseSection" .J.Addition.Cases}}
+// subtraction{{template "caseSection" .J.Subtraction.Cases}}
+// multiplication{{template "caseSection" .J.Multiplication.Cases}}
+// division{{template "caseSection" .J.Division.Cases}}
+// combinedArithmetic{{template "caseSection" .J.CombinedArithmetic.Cases}}
```

Comment:
you used "parsing and numbers" above (with spaces), but `combinedArithmetic` (camel) here. any reason for the difference? Obviously it's not too important since it's a comment, but curious.